internals/overlord/state/state.go

```diff
@@ -172,17 +172,16 @@ type marshalledState struct {
 // marshalledIdentity is used specifically for marshalling to the state
 // database file. Unlike apiIdentity, it should include secrets.
 type marshalledIdentity struct {
-	Access    string                       `json:"access"`
-	Local     *marshalledLocalIdentity     `json:"local,omitempty"`
-	BasicAuth *marshalledBasicAuthIdentity `json:"basicauth,omitempty"`
+	Access string                   `json:"access"`
+	Local  *marshalledLocalIdentity `json:"local,omitempty"`
+	Basic  *marshalledBasicIdentity `json:"basi,omitempty"`
```

Comment:
Just skimming through this PR (not reviewing it yet), but noticed this, so commenting before I forget: basi -> basic